core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1072,31 +1073,226 @@ class KafkaApis(val requestChannel: RequestChannel,
+      )
+    }
+    unconvertedShareAcknowledgeResponse = new ShareAcknowledgeResponse(
+      ShareAcknowledgeResponse.toMessage(Errors.NONE, 0, partitions.entrySet().iterator(), Collections.emptyList())
+    )
+    createResponse()
+  }
+
+  def handleAcknowledgeFromShareFetchRequest (
+                                             request: RequestChannel.Request,
+                                             topicNames : util.Map[Uuid, String],
+                                             sharePartitionManager : SharePartitionManager,
+                                           ) : ShareAcknowledgeResponse = {
+
     val shareFetchRequest = request.body[ShareFetchRequest]
+    val groupId = shareFetchRequest.data().groupId()
+    val clientId = request.header.clientId
+    val memberId = shareFetchRequest.data().memberId()
 
-    if (!config.isShareGroupEnabled) {
-      // The API is not supported when the configuration `group.share.enable` has not been set
-      requestHelper.sendMaybeThrottle(request, shareFetchRequest.getErrorResponse(Errors.UNSUPPORTED_VERSION.exception))
-      CompletableFuture.completedFuture[Unit](())
-      return
+    if (!authHelper.authorize(request.context, READ, GROUP, groupId)) {
+      shareFetchRequest.getErrorAcknowledgeResponse(AbstractResponse.DEFAULT_THROTTLE_TIME, Errors.GROUP_AUTHORIZATION_FAILED.exception)
+    }
+    val interesting = mutable.Map[TopicIdPartition, util.List[SharePartition.AcknowledgementBatch]]()
+    val erroneous = mutable.Map[TopicIdPartition, ShareAcknowledgeResponseData.PartitionData]()
+
+    def getAcknowledgeBatchesFromShareFetchRequest() : mutable.Map[TopicIdPartition, util.List[SharePartition.AcknowledgementBatch]] = {
+
+      val acknowledgeBatchesMap = mutable.Map[TopicIdPartition, util.List[SharePartition.AcknowledgementBatch]]()
+      shareFetchRequest.data().topics().forEach ( topic => {
+
+        if(!topicNames.asScala.contains(topic.topicId)) {
+          topic.partitions.forEach((partition: ShareFetchRequestData.FetchPartition) => {
+            val topicIdPartition = new TopicIdPartition(
+              topic.topicId,
+              new TopicPartition(null, partition.partitionIndex))
+            erroneous +=
+              topicIdPartition -> ShareAcknowledgeResponse.partitionResponse(topicIdPartition, Errors.UNKNOWN_TOPIC_ID)
+          })
+        }
+        else {
+          topic.partitions().forEach ( partition => {
+            val topicIdPartition = new TopicIdPartition(
+              topic.topicId(),
+              new TopicPartition(topicNames.get(topic.topicId()), partition.partitionIndex())
+            )
+            var exceptionThrown = false
+            val acknowledgeBatches = new util.ArrayList[SharePartition.AcknowledgementBatch]()
+            breakable{
+              partition.acknowledgementBatches().forEach( batch => {
+                try {
+                  acknowledgeBatches.add(new SharePartition.AcknowledgementBatch(
```

Comment:
Why is the code calling `SharePartition` functions?  By the initial design, it should be calling functions in `SharePartitionManager`

Reply:
nevermind I see that you are only utilizing a data structure but not the functionality, so it should be fine.